### PR TITLE
Change AbstractSynchronousInstrument to accept a bound factory instead of abstract method

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/DoubleCounterSdk.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
   private DoubleCounterSdk(
       InstrumentDescriptor descriptor, InstrumentAccumulator instrumentAccumulator) {
-    super(descriptor, instrumentAccumulator);
+    super(descriptor, instrumentAccumulator, BoundInstrument::new);
```
Comment (jkwatson) [inline sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/DoubleCounterSdk.java:19]: If all the implementations are the same, can we just inline this into the AbstractSynchronousInstrument constructor and call it a day? Are there ever likely to be other implementations for creating a bound instrument? Or is this the only way to cleanly get the type capture to work right?

Comment (bogdandrutu) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/DoubleCounterSdk.java:19]: They are not the same. BoundInstrument is a different sub-class in the parent caller class

Comment (jkwatson) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/DoubleCounterSdk.java]: oh, you're right. I totally missed that in the github view. That's what I get for not pulling down the code and looking at it fully. 👍

**Review by jkwatson — APPROVED** [review]
👍

File: sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/DoubleCounterSdk.java (hunk 1 above)
Comment (bogdandrutu) [reply sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/DoubleCounterSdk.java]: Thanks :)

**Pull request merged**